contrib/docker/Dockerfile

```diff
@@ -20,9 +23,18 @@ RUN if [ -z "$ZCASH_VERSION" ]; \
     fi; \
     zcashd --version
 
-WORKDIR /srv/zcashd
-RUN mkdir -p /srv/zcashd/.zcash/ \
-    && touch /srv/zcashd/.zcash/zcash.conf
+RUN useradd --home-dir /srv/$ZCASHD_USER \
+            --shell /bin/bash \
+            --create-home \
+            --uid $ZCASHD_UID\
+            $ZCASHD_USER
+
+RUN mkdir -p /srv/$ZCASHD_USER/.zcash/ \
+    && touch /srv/$ZCASHD_USER/.zcash/zcash.conf \
+    && chown -R zcashd /srv/$ZCASHD_USER
```

Comment:
With the latest change, it is possible to run the image using either the root user or the zcashd user (which can be customized during the build using the ZCASHD_USER and ZCASHD_UID variables).